File: server/src/main/java/com/bettercallxiaojin/home/pojo/entity/Notification.java
package com.bettercallxiaojin.home.pojo.entity;

import com.bettercallxiaojin.home.pojo.VO.SimpleUserVO;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    private String id;
    private String userId;
    private Boolean read;
    private Integer type;
    private LocalDateTime createdAt;
    private String targetUserId;
    private String targetId;
    private Integer targetType;
    private String targetContent;
    private String content;
    private String postId;
}

File: web/src/models/notification.ts
interface SimpleUserVO {
  id: string;
  username: string;
  icon: string;
}

export interface NotificationData {
  id: string;
  read: boolean;
  type: number;
  createdAt: string; 
  targetUser: SimpleUserVO | null;
  targetId: string | null;
  targetType: number | null;
  targetContent: string | null;
  content: string | null;
  postId: string | null;
}